### PR TITLE
Generate the acceptance test CA with SAN

### PR DESCRIPTION
**Pull request by ekohl**
Modern tooling prefers the subjectAltName extension and the CN on a certificate is considered deprecated. Some tools even complain about this. However, the -addext command is unavailable on EL7 because openssl is too old.

### CODE REVIEW

File: spec/setup_acceptance_node.pp [hunk 1]
```diff
@@ -45,13 +45,20 @@
 $client_cert = "${directory}/client-cert.pem"
 $client_key = "${directory}/client-key.pem"
 
+# EL7 lacks openssl -addext
+if $facts['os']['release']['major'] == '7' {
+  $ca_cmd = "openssl req -nodes -x509 -newkey rsa:2048 -subj '/CN=${facts['networking']['fqdn']}' -keyout '${ca_key}' -out '${ca_cert}' -days 365",
+} else {
+  $ca_cmd = "openssl req -nodes -x509 -newkey rsa:2048 -subj '/CN=${facts['networking']['fqdn']}' -addext 'subjectAltName = DNS:${facts['networking']['fqdn']}' -keyout '${ca_key}' -out '${ca_cert}' -days 365",
+}
```
Comment (evgeni) [inline spec/setup_acceptance_node.pp:53]:
```suggestion
if $facts['os']['release']['major'] == '7' {
  $ca_cmd = "openssl req -nodes -x509 -newkey rsa:2048 -subj '/CN=${facts['networking']['fqdn']}' -keyout '${ca_key}' -out '${ca_cert}' -days 365"
} else {
  $ca_cmd = "openssl req -nodes -x509 -newkey rsa:2048 -subj '/CN=${facts['networking']['fqdn']}' -addext 'subjectAltName = DNS:${facts['networking']['fqdn']}' -keyout '${ca_key}' -out '${ca_cert}' -days 365"
}
```

File: spec/setup_acceptance_node.pp [hunk 2]
```diff
@@ -45,13 +45,20 @@
 $client_cert = "${directory}/client-cert.pem"
 $client_key = "${directory}/client-key.pem"
 
+# EL7 lacks openssl -addext
+if $facts['os']['release']['major'] == '7' {
+  $ca_cmd = "openssl req -nodes -x509 -newkey rsa:2048 -subj '/CN=${facts['networking']['fqdn']}' -keyout '${ca_key}' -out '${ca_cert}' -days 365"
```
Comment (ehelms) [inline spec/setup_acceptance_node.pp:50]: You could opt to deploy a file with extensions defined in it similar to https://github.com/theforeman/foreman-installer/blob/develop/spec/fixtures/katello-certs-check/extensions.txt

And then point at the extensions when running this command similar to https://github.com/theforeman/foreman-installer/blob/develop/spec/fixtures/katello-certs-check/create_cert.sh#L36

Comment (ekohl) [reply spec/setup_acceptance_node.pp:50]: I considered that, but how long do we plan on supporting EL7? It is mostly needed in the pulp-cli work from #252 but IMHO there's it's sufficient to only check stderr on EL8.

Comment (evgeni) [reply spec/setup_acceptance_node.pp:50]: #252 passes with this change picked on EL8, so I'm happy enough.

(On the other hand, we could consider limiting the number of ways we generate certs in our CI…)

**Review by evgeni — APPROVED** [review]

**Pull request merged**